### PR TITLE
[TT-XLA][PyTorch] Training Workload for Gemma 1.1 2B - DPO

### LINKED ISSUE #225: ☂️ Training DPO workload for google/gemma-1.1-2b-it
Train google/gemma-1.1-2b-it using technique DPO

This training workload is necessary to showcase post-training alignment - same model as 26 but trained on a different task/dataset

### PR DESCRIPTION
**Pull request by mmilosevicTT**
DISCLAIMER: Whole PR was made using Cursor AI prompting.

### Ticket
Closes https://github.com/tenstorrent/tt-blacksmith/issues/225

### Model Training Setting

- **Model:** `Gemma 1.1 2B`
- **Strategy:** DPO (N150/N300/T3K?)
- **Framework:** PyTorch
- **Task:** DPO on Math Preferences

### What's Changed

This PR touches on the following things:
- Refactored `gemma11` experiment to support different training methods (apart from LoRA).
- Implemented DPO training for Gemma 1.1 2B (in 2 stages, sft with LoRA and then DPO on trained model)

### Experiment Results

For completeness here is a train/loss curve of the experiment runs during testing.

TODO

### Checklist
- [x] Main training workflow implementation
- [x] CPU testing
- [ ] GPU testing
- [ ] TT device testing

### CODE REVIEW

**Comment by mmilosevicTT** [comment]
This PR is opened as draft until I manage to unblock it on the failure with `stablehlo.scatter`.

**Comment by mmilosevicTT** [comment]
This is a stale PR. Closing.
